src/cuda.rs

```diff
@@ -0,0 +1,98 @@
+pub type c_char = u8;
+pub type size_t = usize;
+
+cfg_if! {
+    if #[cfg(libc_core_cvoid)] {
+        pub use ::ffi::c_void;
+    } else {
+        // Use repr(u8) as LLVM expects `void*` to be the same as `i8*` to help
+        // enable more optimization opportunities around it recognizing things
+        // like malloc/free.
+        #[repr(u8)]
+        #[allow(missing_copy_implementations)]
+        #[allow(missing_debug_implementations)]
+        pub enum c_void {
+            // Two dummy variants so the #[repr] attribute can be used.
+            #[doc(hidden)]
+            __variant1,
+            #[doc(hidden)]
+            __variant2,
+        }
+    }
+}
+
+extern "C" {
+    /// Syscall to be used whenever the *assert expression produces a `false` value*.
+    ///
+    /// Syscall arguments:
+    /// * `message`: The pointer to the string that should be output.
+    /// * `file`: The pointer to the file name string associated with the assert.
+    /// * `line`: The line number associated with the assert.
+    /// * `function`: The pointer to the function name string associated with the assert.
+    /// * `char_size`: The size in bytes of the characters contained in the provided strings. The only supported character size is 1.
+    ///
+    /// Source:
+    /// [PTX Interoperability](https://docs.nvidia.com/cuda/ptx-writers-guide-to-interoperability/index.html#system-calls).
+    pub fn __assertfail(
+        message: *const c_char,
+        file: *const c_char,
+        line: i32,
```

Comment:
line is `unsigned` in clang, which would map to `c_uint`, but the syscall docs state that this should indeed be a 32-bit integer. Maybe use either `u32` here, or define a `c_uint = u32` ?

Reply:
Surprisingly, the documentation states incorrect information while Clang has a correct implementation! I tried to call it with `-1` as `line`, and it printed it as `uint(-1)`!

```
src/main.rs:4294967295: (null): block: [1,0,0], thread: [1,0,0] Assertion `Panicked` failed.
```